observability/src/main/java/io/grpc/observability/logging/GcpLogSink.java

```diff
@@ -78,13 +97,17 @@ public void write(GrpcLogRecord logProto) {
     try {
       GrpcLogRecord.EventType event = logProto.getEventType();
       Severity logEntrySeverity = getCloudLoggingLevel(logProto.getLogLevel());
+      MonitoredResource resource = getResource(locationTags);
       // TODO(vindhyan): make sure all (int, long) values are not displayed as double
-      LogEntry grpcLogEntry =
+      LogEntry.Builder grpcLogEntryBuilder =
           LogEntry.newBuilder(JsonPayload.of(protoToMapConverter(logProto)))
               .setSeverity(logEntrySeverity)
               .setLogName(DEFAULT_LOG_NAME)
-              .setResource(MonitoredResource.newBuilder("global").build())
-              .build();
+              .setResource(resource);
+      if ((customTags != null) && !customTags.isEmpty()) {
+        grpcLogEntryBuilder.setLabels(customTags);
+      }
+      LogEntry grpcLogEntry = grpcLogEntryBuilder.build();
       synchronized (this) {
         logger.log(Level.FINEST, "Writing gRPC event : {0} to Cloud Logging", event);
         gcpLoggingClient.write(Collections.singleton(grpcLogEntry));
```

Comment:
To address our buffering problem, could we do something like this? Keep a "counter" in the `GcpLogSink` object and increment it in this synchronized block. If it reaches N (where N is configurable and set to 100 - say) then call `flush` on `gcpLoggingClient` and reset counter to 0. With this logic the logs will get flushed after every 100 writes (when N is 100). We should think about it

Reply:
Made changes as per your suggestion. PTAL

Reply:
Okay. I was hoping there would be a unit test to verify this logic but I didn't find one in `GcpLogSinkTest`